sympy/polys/polytools.py

```diff
@@ -6218,8 +6216,45 @@ def sqf_list(f, *gens, **args):
     (2, [(x + 1, 2), (x + 2, 3)])
 
     """
-    return _generic_factor_list(f, gens, args, method='sqf')
-
+    options.allowed_flags(args, ['polys'])
+    opt = options.build_options(gens, args)
+    e = together(f)
+    gen_set = set()
+    coeff = 1
+    facs = []
+    for fac in Mul.make_args(e):
+        b, e = (fac.base, fac.exp) if fac.is_Pow else (fac, S.One)
+        if e < 0:
+            raise PolynomialError("a univariate polynomial expected")
+        try:
+            poly, _ = _poly_from_expr(b, opt)
+        except PolificationFailed:
+            coeff *= b
+        else:
+            gen_set.add(poly.gens)
+            if len(gen_set) > 1:
+                raise PolynomialError("a univariate polynomial expected")
+            cf, sqf_facs = poly.sqf_list()
+            coeff *= cf
+            for sqf_fac in sqf_facs:
+                facs.append((sqf_fac[0] , e*sqf_fac[1]))
+    facs = _sorted_factors(facs, 'sqf')
+    if facs:
+        new_facs = [facs[0]]
+        for i in range(1, len(facs)):
+            l = len(new_facs)
+            if facs[i-1][1] == facs[i][1]:
+                new_facs[l-1] = new_facs[l-1][0] * facs[i][0], new_facs[l-1][1]
+            else:
+                new_facs.append(facs[i])
+        if not opt.polys:
+            new_facs = [(f.as_expr(), k) for f, k in new_facs]
+        res = (coeff, new_facs,)
+        return res
+    else:
+        if not opt.polys:
+            facs = [(f.as_expr(), k) for f, k in facs]
```

Comment:
It looks like there are no `facs` at this point.

Reply:
yes, maybe just returning the coeff, facs would be suffice, as the case when facs won't exist is when  the expression is just a constant